packages/react-instantsearch-hooks-dom/src/ui/RangeInput.tsx

```diff
@@ -0,0 +1,174 @@
+   */
+  rootNoRefinement: string;
+  /**
+   * Class names to apply to the form element
+   */
+  form: string;
+  /**
+   * Class name to apply to each label element
+   */
+  label: string;
+  /**
+   * Class names to apply to each input element
+   */
+  input: string;
+  /**
+   * Class names to apply to the minimum input element
+   */
+  inputMin: string;
+  /**
+   * Class names to apply to the maximum input element
+   */
+  inputMax: string;
+  /**
+   * Class names to apply to the separator element
+   */
+  separator: string;
+  /**
+   * Class names to apply to the submit button
+   */
+  submit: string;
+};
+
+export type RangeInputTranslations = {
+  /**
+   * The label of the separator, between the minimum and maximum inputs
+   */
+  separator: string;
+  /**
+   * The label of the submit button
+   */
+  submit: string;
+};
+
+// if the default value is undefined, React considers the component uncontrolled initially, which we don't want 0 or NaN as the default value
+const unsetNumberInputValue = '' as unknown as number;
+
+export function RangeInput({
+  classNames = {},
+  range: { min, max },
+  start: [minValue, maxValue],
+  step = 1,
+  disabled,
+  onSubmit,
+  translations,
+  ...props
+}: RangeInputProps) {
+  const values = {
+    min:
+      minValue !== -Infinity && minValue !== min
+        ? minValue
+        : unsetNumberInputValue,
+    max:
+      maxValue !== Infinity && maxValue !== max
+        ? maxValue
+        : unsetNumberInputValue,
+  };
+
+  const [{ from, to }, setRange] = React.useState({
+    from: values.min,
+    to: values.max,
+  });
+
+  React.useEffect(() => {
+    setRange({ from: values.min, to: values.max });
+  }, [values.min, values.max]);
+
+  const onInput =
+    (key: string) =>
+    ({ currentTarget }: React.SyntheticEvent<HTMLInputElement>) => {
+      const value = Number(currentTarget.value);
+
+      currentTarget.value = value.toString();
+      setRange({ from, to, [key]: value });
+    };
+
+  return (
+    <div
+      {...props}
+      className={cx(
+        cx('ais-RangeInput', classNames.root),
+        disabled &&
+          cx('ais-RangeInput-noRefinement', classNames.rootNoRefinement),
+        props.className
+      )}
+    >
+      <form
+        className={cx('ais-RangeInput-form', classNames.form)}
+        onSubmit={(event) => {
+          event.preventDefault();
+          onSubmit([from, to]);
+        }}
+      >
+        <label className={cx('ais-RangeInput-label', classNames.label)}>
+          <input
+            className={cx(
+              cx('ais-RangeInput-input', classNames.input),
+              cx('ais-RangeInput-input--min', classNames.inputMin)
+            )}
+            type="number"
+            min={min}
+            max={max}
+            value={from}
+            step={step}
+            placeholder={min?.toString()}
+            disabled={disabled}
+            onInput={onInput('from')}
```

Comment:
Why this indirection? We could inline it like this:

```suggestion
            onInput={(event) => {
              setRange({ from: Number(event.currentTarget.value), to });
            }}
```